homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,362 @@
+
+                    if !should_overwrite {
+                        bail!("Aborting... not overwriting existing settings file");
+                    }
+
+                    File::options()
+                        .read(true)
+                        .write(true)
+                        .create_new(false)
+                        .open(&path)
+                        .expect("to open settings file")
+                }
+                err => err.expect("to open settings file"),
+            };
+
+            writeln!(writer, "Writing settings to {:?}", path).expect("to write");
+
+            Ok(Box::new(settings_file))
+        }
+    }
+}
+
+fn handle_key_type(
+    key_type: Option<KeyType>,
+    no_input: bool,
+    _writer: &mut Box<dyn Write>,
+) -> Result<KeyType> {
+    match key_type {
+        Some(key_type) => Ok(key_type),
+        None => {
+            if no_input {
+                bail!("Aborting... cannot prompt for key type in non-interactive mode. Pass `--key-type <KEY_TYPE>` to set it.");
+            }
+
+            let options = vec![KeyType::Ed25519, KeyType::Secp256k1];
+
+            let key_type = Select::new("Select key type", options)
+                .prompt()
+                .map_err(|e| miette!(e))?;
+
+            Ok(key_type)
+        }
+    }
+}
+
+fn handle_key(
+    key_arg: Option<KeyArg>,
+    key_type: KeyType,
+    output_path: PathBuf,
+    no_input: bool,
+    _writer: &mut Box<dyn Write>,
+) -> Result<PubkeyConfig> {
+    let config = match key_arg {
+        None => {
+            if no_input {
+                bail!("Aborting... cannot prompt for key in non-interactive mode. Pass `--key-file <KEY_FILE>` or `--key-seed [<KEY_SEED>]` to configure the key.");
+            }
+
+            let options = vec![
+                PubkeyConfigOption::GenerateFromSeed,
+                PubkeyConfigOption::FromFile,
+            ];
+
+            let pubkey_config_choice =
+                Select::new("How would you like to configure the key?", options)
+                    .prompt()
+                    .map_err(|e| miette!(e))?;
+
+            match pubkey_config_choice {
+                PubkeyConfigOption::GenerateFromSeed => {
+                    let seed = CustomType::<PubkeySeed>::new("Enter the seed for the key")
+                        .with_default(PubkeySeed(rand::thread_rng().gen::<[u8; 32]>()))
+                        .with_default_value_formatter(&|_| "random".to_string())
+                        .with_error_message("Please type a base64 encoding of 32 bytes")
+                        .with_help_message("Base64 encoded 32 bytes")
+                        .prompt()
+                        .map_err(|e| miette!(e))?;
+
+                    PubkeyConfig::GenerateFromSeed(RNGSeed::new(key_type, seed.0))
+                }
+                PubkeyConfigOption::FromFile => {
+                    // HACK: We need to manually instantiate the struct with a custom formatter,
```

Comment:
not sure we need to call this a hack vs constraints on the type/impls?

Reply:
I guess my other Q is if we had to use PathBuf, but I'm guessing yes.

Reply:
@QuinnWilton yeah, seeing your responses, I think we can rip this out easily and go with String.

Reply:
Cool, I'll do that, thanks!